document-store/src/test/java/org/hypertrace/core/documentstore/postgres/PostgresQueryParserTest.java

```diff
@@ -158,16 +170,30 @@ void testParseNonCompositeFilterForJsonField() {
       Assertions.assertEquals("document->>'key1' ~* ?", query);
     }
 
+    {
+      Filter filter = new Filter(Filter.Op.IN, "key1", List.of("abc"));
+      String query = PostgresQueryParser.parseFilter(filter, initParams());
+      Assertions.assertEquals("(document->'key1' ?? ?)", query);
+    }
+
     {
       Filter filter = new Filter(Filter.Op.IN, "key1", List.of("abc", "xyz"));
       String query = PostgresQueryParser.parseFilter(filter, initParams());
-      Assertions.assertEquals("document->>'key1' IN (?, ?)", query);
+      Assertions.assertEquals("(document->'key1' ?? ? OR document->'key1' ?? ?)", query);
+    }
+
+    {
+      Filter filter = new Filter(Op.NOT_IN, "key1", List.of("abc"));
+      String query = PostgresQueryParser.parseFilter(filter, initParams());
+      Assertions.assertEquals("document->'key1' IS NULL OR NOT (document->'key1' ?? ?)", query);
     }
 
     {
       Filter filter = new Filter(Op.NOT_IN, "key1", List.of("abc", "xyz"));
       String query = PostgresQueryParser.parseFilter(filter, initParams());
-      Assertions.assertEquals("document->'key1' IS NULL OR document->>'key1' NOT IN (?, ?)", query);
+      Assertions.assertEquals(
```

Comment:
With regards to document-store, it is better to start with a failing Integration test, followed by the fix and then the unit tests. Otherwise, the query may keep changing requiring modification of unit tests multiple times. 🙂